src/pages/home/report/ReportActionsListItemRenderer.tsx

```diff
@@ -59,36 +51,37 @@ function ReportActionsListItemRenderer({
      * Create a lightweight ReportAction so as to keep the re-rendering as light as possible by
      * passing in only the required props.
      */
-    const action = useMemo(
-        () => ({
-            reportActionID: reportAction.reportActionID,
-            message: reportAction.message,
-            pendingAction: reportAction.pendingAction,
-            actionName: reportAction.actionName,
-            errors: reportAction.errors,
-            originalMessage: reportAction.originalMessage,
-            childCommenterCount: reportAction.childCommenterCount,
-            linkMetadata: reportAction.linkMetadata,
-            childReportID: reportAction.childReportID,
-            childLastVisibleActionCreated: reportAction.childLastVisibleActionCreated,
-            whisperedToAccountIDs: reportAction.whisperedToAccountIDs,
-            error: reportAction.error,
-            created: reportAction.created,
-            actorAccountID: reportAction.actorAccountID,
-            childVisibleActionCount: reportAction.childVisibleActionCount,
-            childOldestFourAccountIDs: reportAction.childOldestFourAccountIDs,
-            childType: reportAction.childType,
-            person: reportAction.person,
-            isOptimisticAction: reportAction.isOptimisticAction,
-            delegateAccountID: reportAction.delegateAccountID,
-            previousMessage: reportAction.previousMessage,
-            attachmentInfo: reportAction.attachmentInfo,
-            childStateNum: reportAction.childStateNum,
-            childStatusNum: reportAction.childStatusNum,
-            childReportName: reportAction.childReportName,
-            childManagerAccountID: reportAction.childManagerAccountID,
-            childMoneyRequestCount: reportAction.childMoneyRequestCount,
-        }),
+    const action: ReportAction = useMemo(
```

Comment:
itsnt better to type it as `Partial<ReportAction>` ?

Reply:
It has all of the required `ReportAction` fields, so maybe it's better to leave it as ReportAction type

Reply:
But why do we still need the `as ReportAction` assertion below?

Reply:
Since ReportAction is a type of `ReportActionBase & OriginalMessage`, where OriginalMessage is a union, TS gives the error without the assertion:
![image](https://github.com/Expensify/App/assets/23176449/43aa9d5a-2ef4-4907-9b9c-5677ecc16e1f)